### PR TITLE
feat: App-specific `ApplicationData` on Skia

### LINKED ISSUE #8910: [Skia] `ApplicationData.Current.LocalFolder` redirects to %appdata%
### Current behavior

`ApplicationData.Current.LocalFolder` redirects to %appdata%, but should redirect to a more app-specific folder.

### Expected behavior

Writing to `ApplicationData.Current.LocalFolder` should be specific to the running app.

## Note for contributors

The behavior for this is currently undefined, as it is mapped to the UWP behavior. We should be mapping to the behavior that `ApplicationData.Current.LocalFolder` shows when running a unpackaged WinAppSDK app for the same API.

### PR DESCRIPTION
**Pull request by MartinZikmund**
GitHub Issue (If applicable): related to #6420, closes #8910

Replaces #12314

## PR Type

What kind of change does this PR introduce?

### <samp>🤖 Generated by Copilot at b807adc</samp>

~~This pull request adds support for the Skia backend to the SamplesApp project, refactors the rendering and extension logic of the Uno.UI.Composition and Uno.UI.Runtime.Skia.Gtk projects, and aligns the app manifest of the SamplesApp.UWP project with the Uno Platform branding. It also renames, deletes, or updates some files and namespaces to improve the code organization and consistency.~~

Nope, you didn't get it this time 😂

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [ ] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: src/Uno.UWP/ApplicationModel/Package.Other.cs [hunk 1]
```diff
@@ -1,106 +1,110 @@
...
+		if (_entryAssembly is null)
 		{
-			_entryAssembly = entryAssembly;
+			return;
 		}
 
-		private void TryParsePackageManifest()
+		var manifest = _entryAssembly.GetManifestResourceStream(PackageManifestName);
+
+		if (manifest is not null)
```
Comment (ajpinedam) [inline src/Uno.UWP/ApplicationModel/Package.Other.cs:61]: nit nit: saving one allocation
```suggestion
		if (_entryAssembly.GetManifestResourceStream(PackageManifestName) is {} manifest)
```

File: src/Uno.UWP/ApplicationModel/Package.cs [hunk 2]
```diff
@@ -8,13 +8,13 @@ public partial class Package
...
-		{
-		}
+		internal Package() => InitializePlatform();
+
+		partial void InitializePlatform();
 
 		public bool IsDevelopmentMode => GetInnerIsDevelopmentMode();
 
-		public PackageId Id => new PackageId();
+		public PackageId Id { get; } = new();
```
Comment (ajpinedam) [inline src/Uno.UWP/ApplicationModel/Package.cs:17]: 🙏

**Comment by MartinZikmund** [comment]
@jeromelaban @Youssef1313 The application data path is now correct and user specific on Windows as well as Unix-based platforms, but my concern is with the `Temp` folder - as `Path.GetTempPath()` just returns `/tmp/` on Linux, which is not really user-specific (as compared to Windows) - https://learn.microsoft.com/en-us/dotnet/api/system.io.path.gettemppath?view=net-7.0&tabs=windows . Also `/tmp` is potentially very temporary, so it could get cleaned up even while the app is running. Is that acceptable? I provided feature flag to allow overriding the path manually, so the developer is still in control

**Comment by Youssef1313** [comment]
@MartinZikmund I think `/var/tmp` would be better on Linux. Tagging @ramezgerges for his view as well.

File: src/Uno.UWP/Storage/ApplicationData.reference.cs [hunk 3]
```diff
@@ -1,23 +1,22 @@
...
+namespace Windows.Storage;
+
+partial class ApplicationData
 {
-	partial class ApplicationData
-	{
-		private static string GetLocalCacheFolder()
-			=> Path.GetTempPath();
+	private static string GetLocalCacheFolder()
+		=> Path.GetTempPath();
```
Comment (ramezgerges) [inline src/Uno.UWP/Storage/ApplicationData.reference.cs:9]: The standard on *nix is to follow FreeDesktop.org's XDG base directory spec. The spec defines environment variables for cache folders, local folders, etc. Usually, apps follow the environment variable if it exists and fall back to very common values if not (.e.g ~/.cache for cache and ~/.local/share for local, etc.). Arch wiki provides a good summary here: https://wiki.archlinux.org/title/XDG_Base_Directory.

**Comment by ramezgerges** [comment]
> @MartinZikmund I think `/var/tmp` would be better on Linux. Tagging @ramezgerges for his view as well.

I'm not sure about the temp directory requirements for Uno specifically, but the vast majority of linux apps/scripts use /tmp directly. Most distros clear /tmp on reboot, and that should be enough for most scenarios. If it isn't, something more persistent like the local folder should probably be used.

**Review by ramezgerges — APPROVED** [review]

File: doc/articles/features/applicationdata.md [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+## Data location on GTK and WPF
+
+In case of GTK and WPF targets the data are stored in application- and user-specific locations on the hard drive. The default path to the various folders depends on the runtime operating system:
+
+**Windows**
+
+- `LocalFolder` - `C:\Users\UserName>\AppData\Local\<Publisher>\<ApplicationName>\LocalState`
+- `RoamingFolder` - `C:\Users\<UserName>\AppData\Local\<Publisher>\<ApplicationName>\RoamingState`
+- `LocalCahe` - `C:\Users\<UserName>\AppData\Local\<Publisher>\<ApplicationName>\LocalCache`
+- `TemporaryFolder` - `C:\Users\<UserName>\AppData\Local\<Publisher>\<ApplicationName>\RoamingState`
```
Comment (jeromelaban) [inline doc/articles/features/applicationdata.md:73]: @MartinZikmund Is this right? I would not think the temp state would be in roaming.

Comment (MartinZikmund) [reply doc/articles/features/applicationdata.md:73]: Indeed, copy paste error

**Review by jeromelaban — APPROVED** [review]

**Pull request merged**